client/internal/bootstrap/client_test.go

```diff
@@ -23,382 +24,314 @@ import (
+			setupMocks: func(bootstrapConfig *Config, ctx context.Context, imdsClient *imdsmocks.MockClient,
+				kubeconfigValidator *kubeconfigmocks.MockValidator, serviceClient *akssecuretlsbootstrapv1_mocks.MockSecureTLSBootstrapServiceClient) []byte {
+				kubeconfigValidator.EXPECT().Validate(bootstrapConfig.KubeconfigPath, false).
+					Return(fmt.Errorf("invalid kubeconfig")).Times(1)
 				imdsClient.EXPECT().GetInstanceData(ctx).
-					Return(&imds.VMInstanceData{}, nil).
-					Times(1)
+					Return(&imds.VMInstanceData{}, nil).Times(1)
 				serviceClient.EXPECT().GetNonce(ctx, gomock.Any()).
-					Return(&akssecuretlsbootstrapv1.GetNonceResponse{
-						Nonce: "nonce",
-					}, nil).
-					Times(1)
+					Return(&akssecuretlsbootstrapv1.GetNonceResponse{Nonce: "nonce"}, nil).Times(1)
 				imdsClient.EXPECT().GetAttestedData(ctx, "nonce").
-					Return(&imds.VMAttestedData{
-						Signature: "signedBlob",
-					}, nil).
-					Times(1)
+					Return(&imds.VMAttestedData{Signature: "signedBlob"}, nil).Times(1)
 				serviceClient.EXPECT().GetCredential(ctx, gomock.Any()).
 					Return(&akssecuretlsbootstrapv1.GetCredentialResponse{
-						EncodedCertPem: "YW55IGNhcm5hbCBwbGVhc3U======",
-					}, nil).
-					Times(1)
-
-				kubeconfigData, err := bootstrapClient.BootstrapKubeletClientCredential(ctx, bootstrapConfig)
-				Expect(kubeconfigData).To(BeNil())
-				Expect(err).ToNot(BeNil())
-				expectBootstrapErrorWithType(err, ErrorTypeGetCredentialFailure)
-				Expect(err.Error()).To(ContainSubstring("failed to decode cert data from bootstrap server"))
-			})
-		})
-
-		When("bootstrap server can generate a credential", func() {
-			It("should return a new kubeconfig object referencing the new credential", func() {
+						EncodedCertPem: "YW55IGNhcm5hbCBwbGVhc3U======", // base64 encoded invalid PEM
+					}, nil).Times(1)
+				return nil
+			},
+			expectedError: &BootstrapError{
+				errorType: ErrorTypeGetCredentialFailure,
+				inner:     fmt.Errorf("failed to decode cert data from bootstrap server"),
+			},
+			assertKubeconfigData: func(t *testing.T, data *clientcmdapi.Config, _ *Config, _ []byte) {
+				assert.Nil(t, data)
+			},
+		},
+		{
+			name: "bootstrap server can generate a credential",
+			setupMocks: func(bootstrapConfig *Config, ctx context.Context, imdsClient *imdsmocks.MockClient,
+				kubeconfigValidator *kubeconfigmocks.MockValidator, serviceClient *akssecuretlsbootstrapv1_mocks.MockSecureTLSBootstrapServiceClient) []byte {
 				clientCertPEM, _, err := testutil.GenerateCertPEM(testutil.CertTemplate{
 					CommonName:   "system:node:node",
 					Organization: "system:nodes",
 					Expiration:   time.Now().Add(time.Hour),
 				})
-				Expect(err).To(BeNil())
-
+				assert.NoError(t, err)
 				clientCertBlock, rest := pem.Decode(clientCertPEM)
-				Expect(rest).To(BeEmpty())
-
-				kubeconfigValidator.EXPECT().Validate(kubeconfigPath, false).
-					Return(fmt.Errorf("invalid kubeconfig")).
-					Times(1)
+				assert.Empty(t, rest)
```

Comment:
nit: this assertion is probably redundant, you can just omit the `rest` return value with a `_`